File: src/main/java/com/dotronglong/faker/service/handler/NotFoundHandler.java
package com.dotronglong.faker.service.handler;

import com.dotronglong.faker.contract.Handler;
import org.springframework.http.HttpStatus;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class NotFoundHandler implements Handler {
    @Override
    public void handle(HttpServletRequest request, HttpServletResponse response) {
        response.setStatus(HttpStatus.NOT_FOUND.value());
    }
}

File: src/main/kotlin/com/dotronglong/faker/service/handler/NotFoundHandler.kt
package com.dotronglong.faker.service.handler

import com.dotronglong.faker.contract.Handler
import org.springframework.http.HttpStatus
import org.springframework.http.server.reactive.ServerHttpRequest
import org.springframework.http.server.reactive.ServerHttpResponse
import reactor.core.publisher.Mono

class NotFoundHandler : Handler {
    override fun handle(request: ServerHttpRequest, response: ServerHttpResponse): Mono<Void> {
        response.statusCode = HttpStatus.NOT_FOUND
        return Mono.empty()
    }
}